auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -704,4 +750,101 @@ private static void buildParameterizedTypeArguments(CodeBlock.Builder block, Typ
       block.add("$T.class", typeArg);
     }
   }
+
+  private static Optional<ExecutableElement> findDefaultBuilder(TypeElement autoValueClass, TypeElement builderClass) {
+    return ElementFilter.methodsIn(autoValueClass.getEnclosedElements())
+        .stream()
+        .filter(e -> MoreElements.isAnnotationPresent(e, GsonDefaultBuilder.class))
+        .filter(MoreElements.hasModifiers(Modifier.STATIC)::apply)
+        .filter(e -> MoreTypes.equivalence().equivalent(e.getReturnType(), builderClass.asType()))
+        .filter(e -> e.getParameters().isEmpty())
+        .findFirst();
+  }
+
+  /**
+   * Searches the builder for an method has no parameters, returns the autovalue class, is not
+   * private, and is not static. If there is a valid method annotated with {@link GsonBuild}, return
+   * that method. Otherwise, return the method named "build" if present.
+   */
+  private static Optional<ExecutableElement> findBuildMethod(
```

Comment:
This may seem nit-picky, but AutoValue sort of prides itself on not caring what you name your methods.  We shouldn't get in the way of that.